src/oss/langchain/streaming.mdx

```diff
@@ -509,6 +511,386 @@ for await (const [streamMode, chunk] of await agent.stream(
+    {"messages": [input_message]},
+    config=config,  # [!code highlight]
+    stream_mode=["messages", "updates"],
+):
+    interrupts.extend(_process_stream(stream_mode, data))  # [!code highlight]
+```
+```shell title="Output" expandable
+[{'name': 'get_weather', 'args': '', 'id': 'call_GOwNaQHeqMixay2qy80padfE', 'index': 0, 'type': 'tool_call_chunk'}]
+[{'name': None, 'args': '{"ci', 'id': None, 'index': 0, 'type': 'tool_call_chunk'}]
+[{'name': None, 'args': 'ty": ', 'id': None, 'index': 0, 'type': 'tool_call_chunk'}]
+[{'name': None, 'args': '"Bosto', 'id': None, 'index': 0, 'type': 'tool_call_chunk'}]
+[{'name': None, 'args': 'n"}', 'id': None, 'index': 0, 'type': 'tool_call_chunk'}]
+[{'name': 'get_weather', 'args': '', 'id': 'call_Ndb4jvWm2uMA0JDQXu37wDH6', 'index': 1, 'type': 'tool_call_chunk'}]
+[{'name': None, 'args': '{"ci', 'id': None, 'index': 1, 'type': 'tool_call_chunk'}]
+[{'name': None, 'args': 'ty": ', 'id': None, 'index': 1, 'type': 'tool_call_chunk'}]
+[{'name': None, 'args': '"San F', 'id': None, 'index': 1, 'type': 'tool_call_chunk'}]
+[{'name': None, 'args': 'ranc', 'id': None, 'index': 1, 'type': 'tool_call_chunk'}]
+[{'name': None, 'args': 'isco"', 'id': None, 'index': 1, 'type': 'tool_call_chunk'}]
+[{'name': None, 'args': '}', 'id': None, 'index': 1, 'type': 'tool_call_chunk'}]
+Tool calls: [{'name': 'get_weather', 'args': {'city': 'Boston'}, 'id': 'call_GOwNaQHeqMixay2qy80padfE', 'type': 'tool_call'}, {'name': 'get_weather', 'args': {'city': 'San Francisco'}, 'id': 'call_Ndb4jvWm2uMA0JDQXu37wDH6', 'type': 'tool_call'}]
+Tool execution requires approval
+
+Tool: get_weather
+Args: {'city': 'Boston'}
+Tool execution requires approval
+
+Tool: get_weather
+Args: {'city': 'San Francisco'}
+```
+
+We next collect a [decision](/oss/langchain/human-in-the-loop#interrupt-decision-types) for each interrupt. Importantly, the order of decisions must match the order of actions we collected.
+
+To illustrate, we will edit one tool call and accept the other:
+```python
+def _get_interrupt_decisions(interrupt: Interrupt) -> list[dict]:
+    return [
+        {
+            "type": "edit",
+            "edited_action": {
+                "name": "get_weather",
+                "args": {"city": "Boston, U.K."},
+            },
+        }
+        if "boston" in request["description"].lower()
+        else {"type": "approve"}
+        for request in interrupt.value["action_requests"]
+    ]
+
+decisions = []
+for interrupt in interrupts:
+    decisions.extend(_get_interrupt_decisions(interrupt))
+
+decisions
+```
+```shell title="Output"
+[{'type': 'edit',
+  'edited_action': {'name': 'get_weather', 'args': {'city': 'Boston, U.K.'}}},
+ {'type': 'approve'}]
+```
+
+We can then resume by passing a [command](/oss/langchain/human-in-the-loop#responding-to-interrupts) into the same streaming loop:
+```python
+interrupts = []
+for stream_mode, data in agent.stream(
+    Command(resume={"decisions": decisions}),
```

Comment:
We need to show how to resume using interrupt IDs, i.e. so user code supports responding to multiple interrupts in parallel (e.g. two agents running in parallel each using HIL)